src/pds/api_client/preprocess_openapi.py

```diff
@@ -1,16 +1,70 @@
-import json
+import yaml
+import argparse
+import tempfile
+import subprocess
+
+def preprocess(input: dict):
+
+    new_tags_dict = {t['name']: t['name'][3:] for t in input['tags']}
+    for path in input['paths'].keys():
+        tags = input['paths'][path]['get']['tags']
+        new_tags = [new_tags_dict.get(tag, tag) for tag in tags]
+        input['paths'][path]['get']['tags'] = new_tags
+
+    # remove the * object which is confusing for the api generator
+    for k, r in input['components']['responses'].items():
+        del r['content']['*']
+
+    return input
 
 
 def main():
-    with open('swagger.json', 'r') as fi:
-        openapi = json.load(fi)
-        for key, response in openapi['components']['responses'].items():
-            if '*' in response['content']:
-                print(f"remove * content for {key}")
-                response['content'].pop('*')
-
-    with open('preprocessed-swagger.json', 'w') as fo:
-        json.dump(openapi, fo)
-            
+
+    parser = argparse.ArgumentParser(
+                    prog='Process yaml openapi',
+                    description='Remove features unsupported by openapi-generator, python-nextgen language',
+    )
+    parser.add_argument('input_yaml')
+    parser.add_argument('-v', '--version', help="version of the package to be generated")
+
+    args = parser.parse_args()
+
+    with open(args.input_yaml, "r") as stream:
+        try:
+            input = yaml.safe_load(stream)
+            preprocess(input)
+
+        except yaml.YAMLError as exc:
+            print(exc)
+
+    with tempfile.NamedTemporaryFile(mode='w', delete=True) as file:
+        yaml.dump(input, file)
+        clean_cmd = ['rm', '-fr', 'pds', 'test']
+        subprocess.run(clean_cmd)
```

Comment:
Instead of launching a subprocess, Python can do this itself:
```python
import shutil
shutil.rmtree('pds', ignore_errors=True)
shutil.rmtree('test', ignore_errors=True)
```

Reply:
Thanks @nutjob4life